compiler/code-gen/vertex-compiler.cpp

```diff
@@ -1052,17 +1052,24 @@ void compile_switch_var(VertexAdaptor<op_switch> root, CodeGenerator &W) {
   W << temp_var_condition_on_switch << " = " << root->condition() << ";" << NL;
   W << temp_var_matched_with_a_case << " = false;" << NL;
 
-  for (const auto &one_case : root->cases()) {
+  const auto cases = root->cases();
+  const bool default_case_is_the_last = cases.back()->type() == op_default;
```

Comment:
From the syntax point of view, it's possible to have a switch with 0 cases.
```
switch ($cond) {}
```
I wonder if using `back()` is safe in this case.
Maybe add a special case for that?

This kind of code can be a result of code generation, where code generator creates 0-N clauses based on the data.

Reply:
`compile_switch_var()` function is not executed when there is 0 cases in switch, `compile_switch_str()` executed instead. But anyway better be on safe side, fixed.